api/toko-rest-api.yaml

```diff
@@ -1,52 +1,85 @@
----
-swagger: "2.0"
+openapi: 3.0.1
 info:
-  description: "Toko REST API methods description"
-  version: "0.0.1"
-  title: "Toko REST API"
+  title: Toko REST API
+  description: Toko REST API methods description
   contact:
-    email: "dauren.del@gmail.com"
+    email: dauren.del@gmail.com
   license:
-    name: "Apache 2.0"
-    url: "http://www.apache.org/licenses/LICENSE-2.0.html"
-host: "api.toko.kz"
-basePath: "/v1"
+    name: Apache 2.0
+    url: http://www.apache.org/licenses/LICENSE-2.0.html
+  version: 0.0.1
+servers:
+  - url: //api.toko.kz/v1
 tags:
-  - name: "products"
-    description: "Product API to manage user's product catalog"
-  - name: "Users"
-    description: "Operations about user"
-
+  - name: products
+    description: Product API to manage user's product catalog
+  - name: Users
+    description: Operations about user
 paths:
-  # Products API
-  "/products":
+  /products:
     get:
       tags:
-        - "products"
-      summary: "Retrieve all products"
-      operationId: "getProducts"
-      produces:
-        - "application/json"
+        - products
+      summary: Retrieve all products
+      operationId: getProducts
       responses:
-        "200":
+        200:
           description: OK
-          schema:
-            type: array
-            items:
-              $ref: '#/definitions/Product'
-
-definitions:
-  Product:
-     type: object
-     properties:
-       id:
-         type: integer
-         format: int64
-       imageLink:
-         type: string
-         format: uri
-       name:
-         type: string
-       price:
-         type: number
-         format: double
+          content:
+            application/json:
+              schema:
+                type: array
+                items:
+                  $ref: '#/components/schemas/Product'
+  /user:
+    post:
+      tags:
+        - Users
+      summary: Create user
+      description: This method creates a new user.
+      operationId: createUser
+      requestBody:
+        description: Create user request
+        content:
+          '*/*':
+            schema:
+              $ref: '#/components/schemas/CreateUserRequest'
+        required: true
+      responses:
+        default:
+          description: successful operation
+          content: {}
+      x-codegen-request-body-name: body
+components:
+  schemas:
+    Product:
+      type: object
+      properties:
+        id:
+          type: integer
+          format: int64
+        imageLink:
+          type: string
+          format: uri
+        name:
+          type: string
+        price:
+          type: number
+          format: double
+    CreateUserRequest:
+      type: object
+      properties:
+        username:
+          type: string
```

Comment:
We can also add a pattern for` username` to allow Latin letters and dot, underscore and hyphens